lib/src/tasks/coverage/api.dart

```diff
@@ -317,13 +407,79 @@ class CoverageTask extends Task {
     return coverage;
   }
 
+  List<int> observatoryPort;
+
   Future _run() async {
     if (_html && !(await platform_util.isExecutableInstalled('lcov'))) {
       _done.completeError(new MissingLcovException());
       return;
     }
 
-    await _collect();
+    TaskProcess serve;
+
+    if(functional_tests.isNotEmpty){
+      //    String executable = 'selenium-server';
+//    int port = await getOpenPort();
+      List args = ["-port"
+//    ,"${port}"
+      ];
+      args = [];
+      observatoryPort = new List<int>();
+      RegExp _serving = new RegExp("Serving .* on http:\/\/localhost:8080");
+      serve = new TaskProcess('pub', ['serve']);
```

Comment:
Similar note here.. I'd rather this not be something that's different for functional tests. I think there could be some convergence here with @jayudey-wf's work to run all tests during the coverage task via a pub serve process. If all tests were run like this, then we're another step closer to keeping this coverage task "dumber" by not needing to run functional tests differently than other tests